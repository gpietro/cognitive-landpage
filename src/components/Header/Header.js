import React from "react"
import styled from "styled-components"

const Header = styled.div`
  display: flex;
  color: #363c44;
  background-color: #ffcf27;
  align-items: center;
  justify-content: space-between;
`

export default ({ children }) => <Header>{children}</Header>
