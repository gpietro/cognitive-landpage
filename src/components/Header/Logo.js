import React from "react"
import styled from "styled-components"
import Logo from "../../images/assets/logo-cognitive.svg"

const Container = styled.div`
  display: flex;
  align-items: center;
`
export default () => (
  <Container>
    <Logo />
    Cognitive
  </Container>
)
