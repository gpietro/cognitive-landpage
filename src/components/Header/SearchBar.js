import React from "react"
import styled from "styled-components"
import SearchIcon from "../../images/assets/search-icon.svg"
import LineSeparator from "../../images/assets/line-separator.svg"

const SearchBar = styled.div`
  display: flex;
  width: 200px;
  height: 30px;
  color: #fff;
  background-color: #e6bb22;
  border-radius: 3px;
`

const Input = styled.input`
  ::placeholder {
    color: #fff;
  }
  outline: none;
  color: inherit;
  background-color: inherit;
  width: 100%;
  height: 100%;
  border: none;
`
export default () => (
  <SearchBar>
    <SearchIcon />
    <LineSeparator />
    <Input placeholder={"Search for projects or forum topics"} />
  </SearchBar>
)
