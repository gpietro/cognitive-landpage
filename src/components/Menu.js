import React from "react"
import styled from "styled-components"
import { Link } from "gatsby"

const Menu = styled.ul`
  display: flex;
  list-style-type: none;
`

export default ({ menuItems = [] }) => (
  <Menu>
    {menuItems.map(menuItem => (
      <li>
        <Link to="#">{menuItem}</Link>
      </li>
    ))}
  </Menu>
)
