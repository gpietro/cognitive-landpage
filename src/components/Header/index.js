import Logo from "./Logo"
import Header from "./Header"
import SearchBar from "./SearchBar"

export default Header

export { Logo, SearchBar }
