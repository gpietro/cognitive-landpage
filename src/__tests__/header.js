import React from "react"
import renderer from "react-test-renderer"
import Header, { Logo, SearchBar } from "../components/Header"
import Menu from "../components/Menu"

describe("Header", () => {
  it("Render empty header", () => {
    const tree = renderer.create(<Header />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("Render full header", () => {
    const tree = renderer
      .create(
        <Header>
          <Logo />
          <SearchBar />
          <Menu menuItems={["Learn", "Projects", "Forums"]} />
        </Header>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
