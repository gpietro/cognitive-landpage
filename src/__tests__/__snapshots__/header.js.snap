// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Render empty header 1`] = `
<div
  className="sc-bwzfXH boNbIZ"
/>
`;

exports[`Header Render empty header 2`] = `
<div
  className="sc-bwzfXH boNbIZ"
>
  <div
    className="sc-bdVaJa hKAetY"
  >
    <test-file-stub />
    Cognitive
  </div>
  <div
    className="sc-bxivhb gZdxfa"
  >
    <test-file-stub />
    <test-file-stub />
    <input
      className="sc-ifAKCX kSsUJV"
      placeholder="Search for projects or forum topics"
    />
  </div>
  <ul
    className="sc-htpNat kqiIjl"
  >
    <li>
      Learn
    </li>
    <li>
      Projects
    </li>
    <li>
      Forums
    </li>
  </ul>
</div>
`;

exports[`Header Render full header 1`] = `
<div
  className="sc-bwzfXH boNbIZ"
>
  <div
    className="sc-bdVaJa hKAetY"
  >
    <test-file-stub />
    Cognitive
  </div>
  <div
    className="sc-bxivhb gZdxfa"
  >
    <test-file-stub />
    <test-file-stub />
    <input
      className="sc-ifAKCX kSsUJV"
      placeholder="Search for projects or forum topics"
    />
  </div>
  <ul
    className="sc-htpNat kqiIjl"
  >
    <li>
      Learn
    </li>
    <li>
      Projects
    </li>
    <li>
      Forums
    </li>
  </ul>
</div>
`;

exports[`Header renders correctly 1`] = `
<div
  className="sc-bdVaJa hDAbLH"
/>
`;
